src/transformState.js

```diff
@@ -6,6 +6,42 @@
  */
 function transformState(state, actions) {
   // write code here
+  const robotActions = Object.assign({}, actions);
+  const a = Object.keys(robotActions);
+
+  for (let actionNumber = 0; actionNumber < a.length; actionNumber++) {
+    const robotAction = robotActions[actionNumber];
+
+    switch (robotAction.type) {
+      case 'addProperties':
+
+        const keys = Object.keys(robotAction.extraData);
+
+        for (let j = 0; j < keys.length; j++) {
```

Comment:
he for...in loop in the addProperties case can be replaced with a simpler for...of loop that iterates over the keys of the extraData object directly.
sth like `for (const key of Object.keys(action.extraData)) {`